src/pybuffer.jl

```diff
@@ -37,6 +37,13 @@ mutable struct PyBuffer
     end
 end
 
+"""
+`pydecref(o::PyBuffer)`
+Release the reference to buffer `o`
+N.b. As per https://docs.python.org/3/c-api/buffer.html#c.PyBuffer_Release,
+It is an error to call this function on a PyBuffer that was not obtained via
+the python c-api function `PyObject_GetBuffer()`
```

Comment:
Note that this docstring (which is "correct" according to cpython documentation) is in contradiction to the comment _`a no-op if o.obj is already NULL`_ just below in the source code (which reflects the actual cpython implementation).

IIUC, what is written in the comment is actually important for understanding why [`PyBuffer(o::Union{PyObject,PyPtr}, flags=PyBUF_SIMPLE)`](https://github.com/JuliaPy/PyCall.jl/pull/487/files#diff-4ca60bdd8276bb88144aeefa48f65a6dR118) is correct since it essentially does `pydecref(PyBuffer())`. If that's the case, maybe clarify this point in the docstring?

Reply:
Good point. Updated.

Reply:
Good point. Updated.

Reply:
Good point. Updated.

Reply:
Lol - github seems to be having some sync issues. At least I've made my agreement very clear 😄